lib/avromatic/model/raw_serialization.rb

```diff
@@ -29,6 +29,9 @@ def recursive_serialize(value, name: nil, member_types: nil)
             if value.is_a?(Avromatic::Model::Attributes)
               hash = value.value_attributes_for_avro
               if Avromatic.use_custom_datum_writer
+                unless value.class.config.mutable
+                  hash = { Avromatic::IO::ENCODING_PROVIDER => value }
```

Comment:
Makes sense. As discussed offline let's just include a comment on why it makes sense to always return a Hash.